File: WIP - Progetto/app/src/main/java/it/wip/database/model/Chapter.kt
package it.wip.database.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(tableName = "chapter",
        foreignKeys = [
            ForeignKey(
                entity = Story::class,
                parentColumns = ["id"],
                childColumns = ["story"],
                onUpdate = ForeignKey.CASCADE,
                onDelete = ForeignKey.CASCADE
            )
        ])
data class Chapter(
    @PrimaryKey(autoGenerate = true)
    val id: Int,
    @ColumnInfo(name = "chapter_name")
    val chapterName: String,
    val time: Int,
    @ColumnInfo(name = "created_on")
    val createdOn: String,
    val story: Int
)

File: wip_flutter/lib/database/model/chapter.dart
class Chapter {

  final int id;
  final String chapterName;
  final int time;
  final String createdOn;
  final int studyTime;
  final int breakTime;
  final String mode;
  final String avatar;
  final int story;

  Chapter({
    required this.id,
    required this.chapterName,
    required this.time,
    required this.createdOn,
    required this.studyTime,
    required this.breakTime,
    required this.mode,
    required this.avatar,
    required this.story
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'chapter_name': chapterName,
      'time': time,
      'created_on': createdOn,
      'study_time': studyTime,
      'break_time': breakTime,
      'mode': mode,
      'avatar': avatar,
      'story': story
    };
  }

}